File: Assembly-FSharp/Assembly-FSharp/Models/Models/Models/Models/Models/Models/Models/Models/Models/Models/TransformModel.fs
namespace Styx.Models

open UnityEngine

[<CreateAssetMenu(fileName = "Transform Model", menuName = "Models/Transform model")>]
type TransformModel() =
    inherit ScriptableObject()

    [<SerializeField>]
    let mutable transform: Transform = null

    member this.Transform
        with get() = transform
        and set(v) = transform <- v

File: Assets/_Scripts/Models/TransformModel.cs
using UnityEngine;

namespace Styx.Models
{
    [CreateAssetMenu(fileName = "Transform Model", menuName = "Models/Transform model")]
    public class TransformModel : ScriptableObject
    {
        [SerializeField]
        private Transform _transform;

        public Transform Transform
        {
            get { return _transform; }
            set { _transform = value; }
        }
    }
}
